src/librustc_passes/consts.rs

```diff
@@ -79,12 +79,12 @@ fn const_is_rvalue_promotable_to_static<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                      .expect("rvalue_promotable_map invoked with non-local def-id");
     let body_id = tcx.hir.body_owned_by(node_id);
     let body_hir_id = tcx.hir.node_to_hir_id(body_id.node_id);
-    tcx.rvalue_promotable_map(def_id).contains_key(&body_hir_id.local_id)
+    tcx.rvalue_promotable_map(def_id).contains(&body_hir_id.local_id)
```

Comment:
I did this exact thing in clippy, and it was very wrong. I'm fairly certain it was wrong here, too.

This checked whether the `local_id` was ever checked for rvalue promotion. Not whether it is rvalue promotable.

Reply:
👍